### ISSUE TITLE
works with newer version guacamole but not specified

### ISSUE THREAD

**Issue by kwibus** [atcomputing/guacamole-trigger#5]

**Comment by adb014** [comment]
This no longer appears to be the case.. At least with a postgresql authentification.. The SQL requests on the login screen return no users and the guacamole login screen does not appear

**Comment by adb014** [comment]
Thanks that would be great.. I'll be testing with a version 1.5.2 of Guacamole, so I'll be able to tell you if it continues to work

**Comment by kwibus** [comment]
want make sure 1.4.0 work correctly before i merge it but i can give you the plugin this way

[guacamole-trigger-1.4.0.tar.gz](https://github.com/atcomputing/guacamole-trigger/files/11788453/guacamole-trigger-1.4.0.tar.gz)

**Comment by kwibus** [comment]
I got it to work. with 1.5.2
when it can't connect to host it will show output of startupscript 
but now  it does not detect at browser side that it can now connect. so it will keep showing startupscript messages.
but if you refresh browser pages it will detect it. and logs you in.

For now i don`t know when i have time to fix this

**Comment by adb014** [comment]
Effectively the startup scripts run and the first time its run it says "Your lab is starting" in green, but guacamole then flags an error with the standard "red" connection lost screen. A reconnect or refresh solves this as the service is now running.

The next time a session is started (after waiting long enough for the STOP_COMMAND to run) the green "Your lab is starting" is no longer displayed, and the "red" conenction error screen is shown immediatemly. However, the container is started and the same symptoms with the browser not recognizing it can connect are seen.

This ok for now, but in production it'll be an endless source of complaints ;-( 

In any case thanks for your reactivity, and here hoping you'll think of the solution to the browser detecting it can connect..

**Comment by kwibus** [comment]
It should no work properly with this version.

[guacamole-trigger-1.4.0.tar.gz](https://github.com/atcomputing/guacamole-trigger/files/11832112/guacamole-trigger-1.4.0.tar.gz)

**Comment by kwibus** [comment]
I will make it it soon. 
but before that i will update the documentation at least

**Comment by adb014** [comment]
Finally I'll have to keep my dnsmasq because 

* If guacd and the started containers are in the same network and therefore the docker address resolution will resolve the container name to the IP of the container
* If guacd and the started containers aren't in the same network, and so the containers XRDP/VNC port needs to be exposed on the gateway of the same network as the guacd container. As "guacamole trigger" uses the hostname to track containers we can't use the IP address as the hostname and as docker address resolution won't work between two different docker networks we absolutely need a DNS resolution setup 

Fine, I give up here :-( This is rather a large gotcha so I'll docuemnt it here

**Comment by kwibus** [comment]
> As "guacamole trigger" uses the hostname to track containers we can't use the IP address as the hostname
 
You can use IP Address as the hostname. 
The only thing you have to watch out for is that if you have multiple ways you refer to same machine. (multiple ip, ip and hostname, multiple hostnames/fqdns) its sees them as multiple machines. which might tricker a stop command on a host. while it has still connections under a different "hostname/ (ip)"

**Comment by adb014** [comment]
Yes but as they are in different docker networks they must be "exposed" and they'll all be exposed on the gateway address of the network where the guacd service is running

**Comment by adb014** [comment]
Any news on an official 1.4 release ?

**Comment by kwibus** [comment]
its here ;)
and also 1.6
